### PR TITLE
[photon-core] 2D Detection data accuracy

### PR DESCRIPTION
**Pull request by amquake**
- [x] Use calibration data for 2d target info when available (principal point, FOV)
- [x] Correct perspective distortion in 2d yaw/pitch info
  - <details><summary>Explanation:</summary>
    The pitch traditionally calculated from pixel offsets do not correctly account for non-zero values of yaw because of perspective distortion (not to be confused with lens distortion)-- for example, the pitch angle is naively calculated as: <pre>pitch = arctan(pixel y offset / focal length y)</pre>
     However, using focal length as a side of the associated right triangle is not correct when the pixel x value is not 0, because the distance from this pixel (projected on the x-axis) to the camera lens increases. Projecting a line back out of the camera with these naive angles will not intersect the 3d point that was originally projected into this 2d pixel. Instead, this length should be: <pre>focal length y ⟶ (focal length y / cos(arctan(pixel x offset / focal length x)))</pre>
    This cannot, however, account for perspective distortion in the calculation of the target center (averaging the contour corners).
  </details>
- ~~Undistort detected target corners when calibration data is available?~~
  - ~~We currently only do this for AprilTag detections when passing to pose estimation. Not sure how correct it would be for all targets or if it's even helpful. Relevant: #643~~ Nah
- [x] Verified on hardware

### CODE REVIEW

**Comment by amquake** [comment]
@mcm001 Thoughts on inverting 2d target yaw / pitch here to match wpilib?

**Comment by mcm001** [comment]
I'm not sure. I see arguments on both sides. If we yell very loudly that it changed, it'll be fine? Like it needs to be at the top of our release notes.

Or @Bankst should we leave it as is?

**Comment by mdurrani808** [comment]
> I'm not sure. I see arguments on both sides. If we yell very loudly that it changed, it'll be fine? Like it needs to be at the top of our release notes.
> 
> Or @Bankst should we leave it as is?

I say we leave it as is and maintain parity with what LL has (large amount of users are coming from there or have experience with LL system anyways).

File: photon-core/src/test/java/org/photonvision/vision/target/TargetCalculationsTest.java [hunk 1]
```diff
@@ -49,43 +57,104 @@ public class TargetCalculationsTest {
...
+                Arguments.of(30, 45),
+                Arguments.of(-45, -20));
+    }
 
-        assertEquals(Math.toDegrees(truePitch) * -1, pitch, 0.025, "Pitch not as expected");
+    private static double[] testCameraMatrix = {240, 0, 320, 0, 240, 320, 0, 0, 1};
+
+    @ParameterizedTest
+    @MethodSource("testYawPitchCalcArgs")
+    public void testYawPitchCalc(double yawDeg, double pitchDeg) {
```
Comment (amquake) [inline photon-core/src/test/java/org/photonvision/vision/target/TargetCalculationsTest.java:123]: Added a test verifying the calculated target yaw/pitch. This fails with the previous calculation, but passes with this PR.

**Review by mcm001 — APPROVED** [review]

**Pull request merged**